File: manual/data/python/tests/test_nltk_stem_PorterStemmer.py
import unittest
from function_nltk_stem_PorterStemmer import Stemmer

class TestPorterStemmer(unittest.TestCase):

    def test_stem_running(self):
        stemmer = Stemmer()  # 实例化 Stemmer 类
        result = stemmer.test_stem('running')  # 调用类方法
        self.assertEqual(result, 'run')

    def test_stem_played(self):
        stemmer = Stemmer()
        result = stemmer.test_stem('played')
        self.assertEqual(result, 'play')

    def test_stem_dogs(self):
        stemmer = Stemmer()
        result = stemmer.test_stem('dogs')
        self.assertEqual(result, 'dog')

    def test_stem_empty_string(self):
        stemmer = Stemmer()
        result = stemmer.test_stem('')
        self.assertEqual(result, '')

    def test_stem_word_with_special_characters(self):
        stemmer = Stemmer()
        result = stemmer.test_stem('$#@!word')
        self.assertEqual(result, '$#@!word')


if __name__ == '__main__':
    unittest.main()

File: manual/data/java/tests/test_nltk_stem_PorterStemmer.java
package com.example;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class test_nltk_stem_PorterStemmer {

    @Test
    public void test_stem_running() {
        function_nltk_stem_PorterStemmer stemmer = new function_nltk_stem_PorterStemmer();
        String result = stemmer.test_stem("running");
        assertEquals("run", result);
    }

    @Test
    public void test_stem_played() {
        function_nltk_stem_PorterStemmer stemmer = new function_nltk_stem_PorterStemmer();
        String result = stemmer.test_stem("played");
        assertEquals("plai", result);
    }

    @Test
    public void test_stem_dogs() {
        function_nltk_stem_PorterStemmer stemmer = new function_nltk_stem_PorterStemmer();
        String result = stemmer.test_stem("dogs");
        assertEquals("dog", result);
    }

    @Test
    public void test_stem_empty_string() {
        function_nltk_stem_PorterStemmer stemmer = new function_nltk_stem_PorterStemmer();
        String result = stemmer.test_stem("");
        assertEquals("", result);
    }

    @Test
    public void test_stem_word_with_special_characters() {
        function_nltk_stem_PorterStemmer stemmer = new function_nltk_stem_PorterStemmer();
        String result = stemmer.test_stem("$#@!word");
        assertEquals("$#@!word", result);
    }
}